apps/scan/src/main/scala/org/lfdecentralizedtrust/splice/scan/store/db/DbScanStore.scala

```diff
@@ -596,48 +596,27 @@ class DbScanStore(
   ): Future[BigDecimal] =
     waitUntilAcsIngested {
       for {
-        result <- ensureAggregated(asOfEndOfRound) { lastAggregatedRound =>
-          // if the asOfEndOrRound is the latest aggregated round, we can use the active_parties for a faster query.
-          if (lastAggregatedRound == asOfEndOfRound) {
-            // using greatest(0, ...) to handle negative balances caused by amulets never expiring.
-            storage.query(
-              // TODO(#800) change to query from round_totals when amulet expiry works again
-              sql"""
-              select sum(greatest(0, rpt.cumulative_change_to_initial_amount_as_of_round_zero - rpt.cumulative_change_to_holding_fees_rate * ($asOfEndOfRound + 1)))
-              from    round_party_totals rpt
-              join    active_parties ap
-              on      rpt.store_id = ap.store_id
-              and     rpt.party = ap.party
-              and     rpt.closed_round = ap.closed_round
-              where   rpt.store_id = $roundTotalsStoreId;
-              """.as[Option[BigDecimal]].headOption,
-              "getTotalAmuletBalanceForLastAggregatedRound",
-            )
-          } else {
-            // fall back if the requested round is not the latest aggregated round.
-            storage.query(
-              // TODO(#800) change to query from round_totals when amulet expiry works again
-              sql"""
-              with most_recent as (
-                select   max(closed_round) as closed_round,
-                party
-                  from     round_party_totals
-                  where    store_id = $roundTotalsStoreId
-                  and      closed_round <= $asOfEndOfRound
-                  group by party
-              )
-              select sum(greatest(0, rpt.cumulative_change_to_initial_amount_as_of_round_zero - rpt.cumulative_change_to_holding_fees_rate * ($asOfEndOfRound + 1)))
-              from   round_party_totals rpt,
-              most_recent mr
-              where  rpt.store_id = $roundTotalsStoreId
-              and    rpt.party = mr.party
-              and    rpt.closed_round = mr.closed_round;
-              """.as[Option[BigDecimal]].headOption,
-              "getTotalAmuletBalanceForEarlierRound",
-            )
-          }
+        result <- ensureAggregated(asOfEndOfRound) { _ =>
+          storage.query(
+            // TODO(#800) change to query from round_totals when amulet expiry works again
+            // the round_total_amulet_balance is sparse and might not have an entry for the requested round
+            // which is why the first entry found <= asOfEndOfRound is used
+            sql"""
```

Comment:
Maybe better to fail with not_found if a early balance is requested, which would signal demand for this (and from that we can decide if we want to re-start scan aggregation for all rounds, or start a separate re-computation process for `round_total_amulet_balance`).

Reply:
We can also document from which round this is available for instance.